sdk/react/src/markers/IconMarker/IconMarker.test.js

```diff
@@ -1,32 +1,67 @@
 // @flow
 
-import * as React from 'react';
-import renderer from 'react-test-renderer';
+import React from 'react';
+import { TouchableWithoutFeedback } from 'react-native';
+import TestRenderer from 'react-test-renderer';
 
 import IconMarker from './IconMarker';
 
 describe('IconMarker', () => {
   it('renders matching snapshot', () => {
-    const tree = renderer.create(
+    const testRenderer = TestRenderer.create(
       <IconMarker
         iconUrl="http://example.com/icon.png"
         imageUrl="http://example.com/image.png"
         onSeek={() => undefined}
       />,
-    ).toJSON();
+    );
 
-    expect(tree).toMatchSnapshot();
+    expect(testRenderer.toJSON()).toMatchSnapshot();
   });
 
-  it('does not render with empty URLs', () => {
-    const tree = renderer.create(
+  it('does not render anything when URLs are empty', () => {
+    const testRenderer = TestRenderer.create(
       <IconMarker
         iconUrl={undefined}
         imageUrl={undefined}
         onSeek={() => undefined}
       />,
-    ).toJSON();
+    );
 
-    expect(tree).toBeNull();
+    expect(testRenderer.toJSON()).toBeNull();
+  });
+
+  it('triggers `onSeek` callback every time when pressed if `imageUrl` is empty', () => {
+    const onSeekMock = jest.fn();
+    const testRenderer = TestRenderer.create(
+      <IconMarker
+        iconUrl="http://example.com/icon.png"
+        imageUrl={undefined}
+        onSeek={onSeekMock}
+      />,
+    );
+
+    testRenderer.root.findByType(TouchableWithoutFeedback).props.onPress();
+
+    expect(onSeekMock).toBeCalled();
+  });
+
+  it('triggers `onSeek` callback when pressed being expanded', () => {
+    const onSeekMock = jest.fn();
+    const testRenderer = TestRenderer.create(
+      <IconMarker
+        iconUrl="http://example.com/icon.png"
+        imageUrl="http://example.com/image.png"
+        onSeek={onSeekMock}
+      />,
+    );
+
+    testRenderer.root.findByType(TouchableWithoutFeedback).props.onPress();
```

Comment:
isn't it better simulate press than call component method directly?

Reply:
How to do it?

Reply:
try `testRenderer.root.findByType(TouchableWithoutFeedback).simulate('press');`

Reply:
`TypeError: wrapper.root.findByType(...).simulate is not a function` it seems to be deprecated as no mentions are present at https://reactjs.org/docs/test-renderer.html Some tutorials I found uses this approach with `props.onClick()`

Reply:
Moved to Enzyme, @ooyala-amohnatkin please review